sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Provider } from '../../providers/provider';
+import {
+  extractMutableReference,
+  extractStructTag,
+  getObjectReference,
+  isSharedObject,
+  isValidSuiAddress,
+  normalizeSuiObjectId,
+  SuiJsonValue,
+  SuiMoveNormalizedType,
+} from '../../types';
+import { bcs, CallArg, ObjectArg } from '../../types/sui-bcs';
+import { MoveCallTransaction } from './txn-data-serializer';
+
+const MOVE_CALL_SER_ERROR = 'Move call argument serialization error:';
+
+const isTypeFunc = (type: string) => (t: any) => typeof t === type;
+
+export class CallArgSerializer {
+  constructor(private provider: Provider) {}
+
+  async serializeMoveCallArguments(
+    txn: MoveCallTransaction
+  ): Promise<CallArg[]> {
+    const normalized = await this.provider.getNormalizedMoveFunction(
+      normalizeSuiObjectId(txn.packageObjectId),
+      txn.module,
+      txn.function
+    );
+    const params = normalized.parameters;
+    // Entry functions can have a mutable reference to an instance of the TxContext
+    // struct defined in the TxContext module as the last parameter. The caller of
+    // the function does not need to pass it in as an argument.
+    if (params.length > 0 && this.isTxContext(params[params.length - 1])) {
+      params.pop();
```

Comment:
I don't love the modification of the returned params here, would rather copying the params into a new array.